Library/Homebrew/cask/auditor.rb

```diff
@@ -53,7 +53,18 @@ def audit
       errors = Set.new
 
       if !language && language_blocks
-        language_blocks.each_key do |l|
+
+        if language_blocks.length > 10 && !@audit_new_cask
+          sample = language_blocks.keys.sample(10)
+          sample_languages = language_blocks.select { |k| sample.include?(k) }
```

Comment:
@bevanjkay I wasn't able to get the shorthand syntax to work because the intersection is between a hash and an array. I'm not sure if there is a more efficient/cleaner way of achieving this.

Reply:
@bevanjkay It looks like you are only ever using the keys here, right? In that case I'd suggest you convert to `language_blocks.keys` first and then use that consistently below rather than repeatedly convert to/loop over keys?